### PR TITLE
Implementation of Message and Request models

### PR DESCRIPTION
**Pull request by TommyP2W**
`Message Model:`
**Message_text** - Contains information within a message sent by a user, with a text limit of up to 1000 words.
**Timestamp** - Defaults to the current configured timezone on Django settings.
**Sender** - Uses a foreign key to a User instance, which will then be manipulated in views that will be committed soon, cannot be null.
**Receiver** - Similar to "Sender", cannot be null.

`Request Model:`
**Driver_id** - Foreign key to a Driver instance, cannot be null.
**Slot** - Foreign key to a Slot instance, cannot be null.
**Arrival and Departure** - Both are datetime fields, which can be manually set.
**Status & Current_status** - Enumerative type with a restricted number of choices, defaults to PENDING.

The Message model is operating as intended, I have not had the chance to test the request model as of yet. However I have created test data on python shell using this model, and this seems to be working correctly.

The two new models have been registered on admin.py. The pull request seems to have deleted Brayden's contributions for some reason however I imagine that this should still be okay to merge.

### CODE REVIEW

**Review by NitroSniper — CHANGES_REQUESTED** [review]
### Issues 
* You have deleted models that was accepted in PR #8 (`Slot` & `Payment`), Please undo those deletes via 
```bash
git cherry-pick 46b237d5100ac7f008ab0eccc6df565b885e116a
```

This will cause a merge conflict most likely and you just have to resolve it. If you need help just msg me while doing it.
#### **Nitpicks**
These are general improvement to the code to help with code smell and future maintenance. 
* attribute `Message_text` does not follow [PEP 8](https://peps.python.org/pep-0008/#method-names-and-instance-variables). perhaps rename it either `text`, `content`,`body`, `content_body` or `message_text`
* `null=False` is default [behaviour](https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.Field.null)
* `choices=current_status.choices` -> `choices=current_status` (I believe)
* class `current_status` should follow [PEP 8](https://peps.python.org/pep-0008/#class-names) -> `Status` or `CurrentStatus`
* You can just set the string for `PENDING`, `APPROVED`, `REJECTED` to be "P", "A", "R". just so it's concise

Everything else looks great! I like how you prematurely add it to admin. and your logic for the model is spot on. Please review the feedback and just send the pull request again!

**Review by TommyP2W — COMMENTED** [review]
I have cherrypicked Braydens previous commit. Please let me know if this is okay.

File: DjangoPMS/backend/models.py [hunk 1]
```diff
@@ -19,6 +23,30 @@ class Driver(BaseUser):
...
 
 
+
+class Message(models.Model):
+        message_text = models.TextField(max_length=1000)
+        timestamp = models.DateTimeField(default=timezone.now)
+        sender = models.ForeignKey(User, on_delete=models.CASCADE, related_name="sender")
+        receiver = models.ForeignKey(User, on_delete=models.CASCADE, related_name="receiver")
+        def __str__(self):
+            return self.Message_text
```
Comment (NitroSniper) [inline DjangoPMS/backend/models.py:33]: Message_text is underfined -> message_text

File: DjangoPMS/backend/models.py [hunk 2]
```diff
@@ -19,6 +23,30 @@ class Driver(BaseUser):
...
+
+class Request(models.Model):
+    driver_id = models.ForeignKey(Driver, on_delete=models.CASCADE, null=False)
+    slot = models.ForeignKey(Slot, on_delete=models.CASCADE, null=False)
+    arrival = models.DateTimeField()
+    departure = models.DateTimeField()
+    class CurrentStatus(models.TextChoices):
+        P = "Pending"
+        A = "Approved"
+        R = "Rejected"
```
Comment (NitroSniper) [inline DjangoPMS/backend/models.py:45]: Meant the other way, My bad
```python
Pending = "P"
Approved = "A"
Rejected = "R"
```

File: DjangoPMS/backend/models.py [hunk 3]
```diff
@@ -19,6 +23,30 @@ class Driver(BaseUser):
...
+        sender = models.ForeignKey(User, on_delete=models.CASCADE, related_name="sender")
+        receiver = models.ForeignKey(User, on_delete=models.CASCADE, related_name="receiver")
+        def __str__(self):
+            return self.Message_text
+
+
+
+class Request(models.Model):
+    driver_id = models.ForeignKey(Driver, on_delete=models.CASCADE, null=False)
+    slot = models.ForeignKey(Slot, on_delete=models.CASCADE, null=False)
```
Comment (NitroSniper) [inline DjangoPMS/backend/models.py:39]: null=False here as well

File: DjangoPMS/backend/models.py [hunk 4]
```diff
@@ -19,6 +23,30 @@ class Driver(BaseUser):
...
+    arrival = models.DateTimeField()
+    departure = models.DateTimeField()
+    class CurrentStatus(models.TextChoices):
+        P = "Pending"
+        A = "Approved"
+        R = "Rejected"
+
+    status = models.CharField(max_length=1, choices=CurrentStatus.choices, default=CurrentStatus.P)
```
Comment (NitroSniper) [inline DjangoPMS/backend/models.py:47]: choices=CurrentStatus instead, and rectify default=CurrentStatus.PENDING

**Review by NitroSniper — CHANGES_REQUESTED** [review]
Good on getting Brayden's code back. just issues with typos

**Review by NitroSniper — APPROVED** [review]
Everything is good, Will accept PR now

**Pull request merged**